runtime/src/serde_snapshot/newer.rs

```diff
@@ -1,86 +1,55 @@
 use {
-    super::{common::UnusedAccounts, *},
+    super::{
+        common::UnusedAccounts,
+        storage::SerializableAccountStorageEntry,
+        utils::{serialize_iter_as_map, serialize_iter_as_seq},
+        *,
+    },
     crate::{ancestors::AncestorsForSerialization, stakes::StakesCache},
     solana_measure::measure::Measure,
     std::{cell::RefCell, sync::RwLock},
 };
 
 type AccountsDbFields = super::AccountsDbFields<SerializableAccountStorageEntry>;
 
-/// the serialized type is fixed as usize
-pub type AppendVecIdSerialized = usize;
-
-// Serializable version of AccountStorageEntry for snapshot format
-#[derive(Clone, Copy, Debug, Default, Eq, PartialEq, Serialize, Deserialize)]
-pub(super) struct SerializableAccountStorageEntry {
-    id: AppendVecIdSerialized,
-    accounts_current_len: usize,
-}
-
-pub trait SerializableStorage {
-    fn id(&self) -> AppendVecIdSerialized;
-    fn current_len(&self) -> usize;
-}
-
-impl SerializableStorage for SerializableAccountStorageEntry {
-    fn id(&self) -> AppendVecIdSerialized {
-        self.id
-    }
-    fn current_len(&self) -> usize {
-        self.accounts_current_len
-    }
-}
-
-#[cfg(RUSTC_WITH_SPECIALIZATION)]
-impl solana_frozen_abi::abi_example::IgnoreAsHelper for SerializableAccountStorageEntry {}
-
-impl From<&AccountStorageEntry> for SerializableAccountStorageEntry {
-    fn from(rhs: &AccountStorageEntry) -> Self {
-        Self {
-            id: rhs.append_vec_id() as AppendVecIdSerialized,
-            accounts_current_len: rhs.accounts.len(),
-        }
-    }
-}
-
 // Deserializable version of Bank which need not be serializable,
 // because it's handled by SerializableVersionedBank.
 // So, sync fields with it!
 #[derive(Clone, Deserialize)]
-pub(crate) struct DeserializableVersionedBank {
```

Comment:
Nothing in here is used outside of this module, so removing the `pub`. Same with `SerializableVersionedBank` below.